vars/pullrequest.groovy

```diff
@@ -0,0 +1,51 @@
+/**
+ * Builds the upstream for a specific project + the project itself (normal PR) + a SONARCLOUD analisys if needed
+ * @param the file with a collection of items following the pattern PROJECT_GROUP/PROJECT_NAME, for example kiegroup/drools
+ * @param currentProject the project to build the stream from, like kiegroup/drools
+ * @param settingsXmlId maven settings xml file id
+ * @param propertiesFileId file that defines the maven goals for each rep
+ */
+def build(List<String> projectCollection, String currentProject, String settingsXmlId, String propertiesFileId, String sonarCloudId, sonarCloudReps = []) {
+    println "Building of project ${currentProject}"
+    println "Project collection: ${projectCollection}"
+
+    util.checkoutProjects(projectCollection, currentProject)
+
+    // Build project tree from currentProject node
+    for (i = 0; currentProject != projectCollection.get(i); i++) {
+        println "Current Upstream Project:" + projectCollection.get(i)
+        util.buildProject(projectCollection.get(i), settingsXmlId, util.getGoals(projectCollection.get(i), propertiesFileId, 'upstream'))
+    }
+
+    println "Build of current project: ${currentProject}"
+    util.buildProject(currentProject, settingsXmlId, util.getGoals(currentProject, propertiesFileId))
+
+    if(sonarCloudReps.contains(currentProject)) {
+        println "SONARCLOUD analysis of : ${currentProject}"
+        buildSonar(currentProject, settingsXmlId, util.getGoals(currentProject, propertiesFileId, 'sonarcloud'), sonarCloudId)
+    } else {
+        println "INFO: ${currentProject} project is not for SONARCLOUD analysis"
+    }
+}
+
+/**
+ *
+ * @param project a string following the pattern PROJECT_GROUP/PROJECT_NAME, for example kiegroup/drools
+ * @param settingsXmlId maven settings xml file id
+ * @param goals maven goals
+ * @param sonarCloudId token for sonarcloud
+ */
+def buildSonar(String project, String settingsXmlId, String goals, String sonarCloudId) {
+    def projectGroupName = util.getProjectGroupName(project)
+    def group = projectGroupName[0]
+    def name = projectGroupName[1]
+
+    println "Building ${group}/${name}"
+    dir("${env.$WORKSPACE}/${group}_${name}") {
+        println "pwd".execute().text
```

Comment:
@Ginxo 
this line 45 gives me `/`
and the next line 46 shows the content of the directory 
I see we are in the root directory ´/´
That's why we get:
´21:04:41  [ERROR] The goal you specified requires a project to execute but there is no POM in this directory (/home/jenkins/workspace/custom/mbiarnes/pullrequest/new-PR-appformer-master.pullrequests/kiegroup_appformer). Please verify you invoked Maven from the correct directory. -> [Help 1]´

There is absolutely no POM in the root directory, but on the other hand it says clearly: (line 43 and 44):
´21:04:38  Building kiegroup/appformer´
and
´21:04:38  Running in /home/jenkins/workspace/custom/mbiarnes/pullrequest/new-PR-appformer-master.pullrequests/kiegroup_appformer´

Reply:
@mbiarnes replaced by `dir("${env.$WORKSPACE}/${group}_${name}")`